### PR TITLE
php81: libxml 2.13 compatibility

### LINKED ISSUE #331126: Build failure: php-simplexml-8.1.2
### Steps To Reproduce

1. build *php-simplexml-8.1.2*

### Build log

```
> PASS SimpleXML: invoking key() after the iterator has already been consumed [tests/key_error.phpt]
┃        > SKIP GHSA-3qrf-m4j2-pcrr (libxml global state entity loader bypass) [tests/libxml_global_state_entity_loader_bypass.phpt] reason: zend-test extension not available
┃        > PASS SimpleXML [profile]: Accessing a simple node [tests/profile01.phpt]
┃        > PASS SimpleXML [profile]: Accessing an array of subnodes [tests/profile02.phpt]
┃        > PASS SimpleXML [profile]: Accessing an attribute [tests/profile03.phpt]
┃        > PASS SimpleXML [profile]: Accessing a namespaced element [tests/profile04.phpt]
┃        > PASS SimpleXML [profile]: Accessing an aliased namespaced element [tests/profile05.phpt]
┃        > PASS SimpleXML [profile]: Accessing a namespaced attribute [tests/profile06.phpt]
┃        > PASS SimpleXML [profile]: Accessing an aliased namespaced attribute [tests/profile07.phpt]
┃        > PASS SimpleXML [profile]: Accessing a namespaced attribute without a namespace [tests/profile08.phpt]
┃        > PASS SimpleXML [profile]: Accessing a namespaced element without a namespace [tests/profile09.phpt]
┃        > PASS SimpleXML [profile]: Accessing two attributes with the same name, but different namespaces [tests/profile10.phpt]
┃        > PASS SimpleXML [profile]: Accessing two elements with the same name, but different namespaces [tests/profile11.phpt]
┃        > PASS SimpleXML [profile]: Accessing namespaced root and non namespaced children [tests/profile12.phpt]
┃        > PASS SimpleXML [profile]: Accessing by namespace prefix [tests/profile13.phpt]
┃        > SKIP SimpleXML [interop]: simplexml_import_dom [tests/simplexml_import_dom.phpt] reason: Required extension missing: dom
┃        > PASS simplexml_load_file() [tests/simplexml_load_file.phpt]
┃        > PASS Incorrectly initialized SimpleXmlElement [tests/simplexml_uninitialized.phpt]
┃        > PASS SPL: SimpleXMLIterator [tests/sxe_001.phpt]
┃        > PASS SPL: SimpleXMLIterator and recursion [tests/sxe_002.phpt]
┃        > PASS SPL: SimpleXMLIterator and getChildren() [tests/sxe_003.phpt]
┃        > PASS SPL: SimpleXMLIterator and overridden iterator methods() [tests/sxe_004.phpt]
┃        > PASS SPL: SimpleXMLIterator and overridden count() [tests/sxe_005.phpt]
┃        > =====================================================================
┃        > TIME END 2024-07-30 16:46:43
┃        >
┃        > =====================================================================
┃        > TEST RESULT SUMMARY
┃        > ---------------------------------------------------------------------
┃        > Exts skipped    :    0
┃        > Exts tested     :   12
┃        > ---------------------------------------------------------------------
┃        >
┃        > Number of tests :  131               127
┃        > Tests skipped   :    4 (  3.1%) --------
┃        > Tests warned    :    0 (  0.0%) (  0.0%)
┃        > Tests failed    :    1 (  0.8%) (  0.8%)
┃        > Tests passed    :  126 ( 96.2%) ( 99.2%)
┃        > ---------------------------------------------------------------------
┃        > Time taken      :    1 seconds
┃        > =====================================================================
┃        >
┃        > =====================================================================
┃        > FAILED TEST SUMMARY
┃        > ---------------------------------------------------------------------
┃        > Bug #79971 (special character is breaking the path in xml function) [tests/bug79971_1.phpt]
┃        > =====================================================================
┃        > make: *** [Makefile:136: test] Error 1
┃        For full logs, run 'nix log /nix/store/qhz26ig0n0d43xapgjwp2rj1nllfi133-php-simplexml-8.1.29.drv'.
```

### Additional context

This happen after latest flake update

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.1, NixOS, 24.11 (Vicuna), 24.11.20240727.b73c222`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"home-manager-23.11.tar.gz, nixos-23.11"`
 - nixpkgs: `/nix/store/qxf6anli54ij0q1sdlnlgx9hyl658a4v-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
## Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/331126

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -617,7 +623,20 @@ in {
           configureFlags = [
             "--enable-simplexml"
           ];
-          patches = lib.optionals (lib.versions.majorMinor php.version == "8.2" && lib.versionOlder php.version "8.2.22") [
+          patches = lib.optionals (lib.versions.majorMinor php.version == "8.1") [
+            (fetchpatch {
```
Comment (drupol) [inline pkgs/top-level/php-packages.nix:627]: I like the idea of having comments before each patch so we know if we have to keep them or remove them at the next bump. Can you please add a single comment line like you did in pkgs/development/interpreters/php/8.1.nix ?

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Thank you !!!

File: pkgs/development/interpreters/php/8.1.nix [hunk 2]
```diff
@@ -5,12 +5,28 @@ let
...
       })
+      # Fix tests with libxml2 2.12
+      (fetchpatch {
+        url = "https://github.com/php/php-src/commit/061058a9b1bbd90d27d97d79aebcf2b5029767b0.patch";
+        hash = "sha256-0hOlAG+pOYp/gUU0MUMZvzWpgr0ncJi5GB8IeNxxyEU=";
+        excludes = [
+          "NEWS"
+        ];
+      })
+      # Backport of PHP_LIBXML_IGNORE_DEPRECATIONS_START and PHP_LIBXML_IGNORE_DEPRECATIONS_END
```
Comment (jtojnar) [inline pkgs/development/interpreters/php/8.1.nix:22]: What commit does this come from?

Comment (piotrkwiecinski) [reply pkgs/development/interpreters/php/8.1.nix:22]: @jtojnar I'll PR actual commit patch instead of file. I'm just waiting for build to finish.

**Pull request merged**